File: frontend/src/types/docvault.ts
export type DocumentStatus = 
  | 'uploaded'
  | 'pending_approval'
  | 'action_required'
  | 'verified'
  | 'submitted';

export interface BucketResponse {
  id: string;
  name: string;
  company_id: string;
  created_by: string;
  created_at: string;
  updated_at: string;
}

export interface BucketCreate {
  name: string;
}

export interface DocumentVersionResponse {
  id: string;
  document_id: string;
  original_filename: string;
  mime_type: string;
  size_bytes: number;
  checksum: string;
  uploaded_by: string;
  uploaded_at: string;
  version_number: number;
}

export interface DocumentResponse {
  id: string;
  company_id: string;
  current_version_id: string | null;
  bucket_id: string | null;
  status: DocumentStatus;
  title: string;
  doc_type_id: string | null;
  tags: string[];
  is_editable: boolean;
  created_by: string;
  created_at: string;
  updated_at: string;
  versions: DocumentVersionResponse[];
}

export interface DocumentUpdate {
  status?: DocumentStatus | null;
  bucket_id?: string | null;
  tags?: string[] | null;
  is_editable?: boolean | null;
}

File: app/schemas/docvault.py
import uuid
from datetime import datetime
from typing import Optional, List
from pydantic import BaseModel, Field

from app.models.docvault import DocumentStatus


class BucketCreate(BaseModel):
    name: str = Field(..., max_length=255)


class BucketResponse(BaseModel):
    id: uuid.UUID
    name: str
    company_id: uuid.UUID
    created_by: Optional[uuid.UUID]
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}


class DocumentVersionResponse(BaseModel):
    id: uuid.UUID
    document_id: uuid.UUID
    original_filename: str
    mime_type: str
    size_bytes: int
    checksum: str
    uploaded_by: Optional[uuid.UUID]
    uploaded_at: datetime
    version_number: int

    model_config = {"from_attributes": True}


class DocumentResponse(BaseModel):
    id: uuid.UUID
    company_id: uuid.UUID
    current_version_id: Optional[uuid.UUID]
    bucket_id: Optional[uuid.UUID]
    status: DocumentStatus
    title: str
    doc_type_id: Optional[uuid.UUID]
    tags: List[str]
    is_editable: bool
    created_by: Optional[uuid.UUID]
    created_at: datetime
    updated_at: datetime
    versions: List[DocumentVersionResponse] = []

    model_config = {"from_attributes": True}


class DocumentUpdate(BaseModel):
    status: Optional[DocumentStatus] = None
    bucket_id: Optional[uuid.UUID] = None
    tags: Optional[List[str]] = None
    is_editable: Optional[bool] = None
